python/pyspark/sql/session.py

```diff
@@ -1023,6 +1023,20 @@ def prepare(obj: Any) -> Any:
 
         if isinstance(data, RDD):
             rdd, struct = self._createFromRDD(data.map(prepare), schema, samplingRatio)
+        elif isinstance(data, list) and schema is None:
+            # Wrap each element with a tuple if there is any scalar in the list
+
+            has_scalar = any(
+                isinstance(x, str)
+                or (
+                    not isinstance(x, Sized)
```

Comment:
How does this work in case of nested types? e.g., `spark.createDataFrame([(1,), (2,)], schema='struct<a: int>')`. We should probably check how the Scala side works, and match the behaviour.